platforms/r3-corda/charts/notary-initial-registration/templates/job.yaml

```diff
@@ -133,6 +133,41 @@ spec:
                   COUNTER=`expr "$COUNTER" + 1`
                 fi
             done
+
+            COUNTER=1
+            while [ "$COUNTER" -lt {{ $.Values.healthcheck.readinessthreshold }} ]
+            do
+                if [ -e nodekeystore.jks ] && [ -e sslkeystore.jks ] && [ -e truststore.jks ]
+                then
+                  # get nodekeystore.jks from vault
+                  LOOKUP_SECRET_RESPONSE=$(curl -sS --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR}/v1/${CERTS_SECRET_PREFIX}/nodekeystore | jq -r 'if .errors then . else . end')
+                  TLS_NODEKEYSTORE=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data[ "nodekeystore.jks" ]' 2>&1)
+
+                  # get sslkeystore.jks from vault
+                  LOOKUP_SECRET_RESPONSE=$(curl -sS --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR}/v1/${CERTS_SECRET_PREFIX}/sslkeystore | jq -r 'if .errors then . else . end')
+                  TLS_SSLKEYSTORE=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data[ "sslkeystore.jks" ]' 2>&1)
+
+                  # get truststore.jks from vault
+                  LOOKUP_SECRET_RESPONSE=$(curl -sS --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR}/v1/${CERTS_SECRET_PREFIX}/truststore | jq -r 'if .errors then . else . end')
+                  TLS_TRUSTSTORE=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data[ "truststore.jks" ]' 2>&1)  
+                fi
+
+                if [ "$TLS_NODEKEYSTORE" == "null" ] || [ "$TLS_SSLKEYSTORE" == "null" ] || [ "$TLS_TRUSTSTORE" == "null" ] || [[ "$TLS_NODEKEYSTORE" == "parse error"* ]] || [[ "$TLS_SSLKEYSTORE" == "parse error"* ]] || [[ "$TLS_TRUSTSTORE" == "parse error"* ]]
+                then
+                  echo "certificates are not ready, sleeping for 5s"
+                  sleep {{ $.Values.healthcheck.readinesscheckinterval }}
+                  COUNTER=`expr "$COUNTER" + 1`
+                else
+                  echo "cryto materials are successfully stored in vault"
+                  break
+                fi
+            done
+            
+            if [ "$COUNTER" -ge {{ $.Values.healthcheck.readinessthreshold }} ]
+            then
+              echo "Retry attempted $COUNTER times, certificates have not been saved"
+              break
```

Comment:
You need to put exit 1 when the threshold limit exceeds for the container to fail
Use proper variable names for retry and retry threshold
Use the same while loop block to save in vault and check as well